### PR TITLE
[DataLake][SetExpiry]Set Expiry of DataLake File

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py [hunk 1]
```diff
@@ -788,13 +784,14 @@ def get_blob_properties(self, **kwargs):
...
                 modified_access_conditions=mod_conditions,
-                cls=deserialize_blob_properties,
+                cls=kwargs.pop('cls', None) or deserialize_blob_properties,
                 cpk_info=cpk_info,
                 **kwargs)
         except StorageErrorException as error:
             process_storage_error(error)
         blob_props.name = self.blob_name
-        blob_props.container = self.container_name
+        if isinstance(blob_props, BlobProperties):
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:793]: Please use 
```
try: 
 ## 
except AttributeError:
 ###
```
If mypy is complaining, use cast()

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py [hunk 2]
```diff
@@ -483,13 +480,14 @@ async def get_blob_properties(self, **kwargs):
...
                 modified_access_conditions=mod_conditions,
-                cls=deserialize_blob_properties,
+                cls=kwargs.pop('cls', None) or deserialize_blob_properties,
                 cpk_info=cpk_info,
                 **kwargs)
         except StorageErrorException as error:
             process_storage_error(error)
         blob_props.name = self.blob_name
-        blob_props.container = self.container_name
+        if isinstance(blob_props, BlobProperties):
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py:489]: same here 

https://docs.python.org/2/glossary.html#term-eafp

interesting read about why - Easier to ask for forgiveness than permission

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py [hunk 3]
```diff
@@ -239,8 +239,22 @@ def get_file_properties(self, **kwargs):
                 :dedent: 4
                 :caption: Getting the properties for a file.
         """
-        blob_properties = self._get_path_properties(**kwargs)
-        return FileProperties._from_blob_properties(blob_properties)  # pylint: disable=protected-access
+        return self._get_path_properties(cls=FileProperties._deserialize_file_properties, **kwargs)  # pylint: disable=protected-access
+
+    def set_file_expiry(self, expiry_options, expires_on=None, **kwargs):
+        # type: (**Any) -> None
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py:245]: wrong type hint

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py [hunk 4]
```diff
@@ -129,33 +129,24 @@ class DirectoryProperties(DictMixin):
...
-        self.name = None
-        self.etag = None
+        self.name = kwargs.get('name')
+        self.etag = kwargs.get('ETag')
         self.deleted = None
-        self.metadata = None
-        self.lease = None
-        self.last_modified = None
-        self.creation_time = None
+        self.metadata = kwargs.get('metadata')
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py:135]: should these all be `kwargs.get('metadata', None)`

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py [hunk 5]
```diff
@@ -326,10 +312,10 @@ class LeaseProperties(BlobLeaseProperties):
     :ivar str duration:
         When a file is leased, specifies whether the lease is of infinite or fixed duration.
     """
-    def __init__(self):
-        self.status = None
-        self.state = None
-        self.duration = None
+    def __init__(self, **kwargs):
+        super(LeaseProperties, self).__init__(
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py:316]: i dont think this __init__ is needed to just call the super

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_file_client_async.py [hunk 6]
```diff
@@ -201,8 +201,22 @@ async def get_file_properties(self, **kwargs):
                 :dedent: 4
                 :caption: Getting the properties for a file.
         """
-        blob_properties = await self._get_path_properties(**kwargs)
-        return FileProperties._from_blob_properties(blob_properties)  # pylint: disable=protected-access
+        return await self._get_path_properties(cls=FileProperties._deserialize_file_properties, **kwargs)  # pylint: disable=protected-access
+
+    async def set_file_expiry(self, expiry_options, expires_on=None, **kwargs):
+        # type: (**Any) -> None
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_file_client_async.py:207]: wrong type hint

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py (hunk 4 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py:135]: That's not needed - `get` already returns None by default.

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py (hunk 5 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py:316]: agree!

**Pull request closed without merging**